src/ApiHandler.ts

```diff
@@ -361,47 +361,63 @@ export default class ApiHandler {
 		}
 	}
 
+	/**
+	 *
+	 * @param hash BlockHash to make call at.
+	 * @param stash: _Stash_ address.
+	 */
 	// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-	async fetchStakingLedger(hash: BlockHash, address: string) {
+	async fetchStakingInfo(hash: BlockHash, stash: string) {
 		const api = await this.ensureMeta(hash);
 
-		const [header, staking] = await Promise.all([
+		const [header, optionController] = await Promise.all([
 			api.rpc.chain.getHeader(hash),
-			api.query.staking.ledger.at(hash, address),
+			api.query.staking.bonded.at(hash, stash), // Option<AccountId> representing the controller
 		]);
 
 		const at = {
 			hash,
 			height: header.number.toNumber().toString(10),
 		};
 
-		if (staking.isNone) {
-			// Address is not a Controller, no need to look up slashing spans.
+		if (optionController.isNone === true) {
+			// if bonded.isNone, `stash` address param is not a stash.
 			return {
 				at,
-				staking: {},
+				controller: null,
+				rewardDestination: null,
 				numSlashingSpans: null,
+				staking: {},
 			};
 		}
 
-		const ledger = staking.unwrap(); // should always work if staking.isSome
-		const slashingSpans = await api.query.staking.slashingSpans.at(
-			hash,
-			ledger.stash
-		);
+		// we can infer (optionBonded.isSome === true) and safely unwrap
+		const controller = optionController.unwrap();
 
-		let numSlashingSpans;
-		if (slashingSpans.isSome) {
-			const span = slashingSpans.unwrap();
-			numSlashingSpans = span.prior.length + 1;
-		} else {
-			numSlashingSpans = 0;
-		}
+		const [
+			optionStakingLedger,
+			rewardDestination,
+			optionSlashingSpans,
+		] = await Promise.all([
+			await api.query.staking.ledger.at(hash, controller),
+			await api.query.staking.payee.at(hash, stash),
+			await api.query.staking.slashingSpans.at(hash, stash),
+		]);
+
+		// should always work because we know that we have a bonded pair but just
+		// in case we return a default value of null so we know to handle later
+		const stakingLedger = optionStakingLedger.unwrapOr(null);
```

Comment:
Would it be appropriate to check for `null` and throw an error if that's the case?

Reply:
Check out my modification to the above if statement and lmk if that makes sense.